persistence/provider/memory/eventstore.go

```diff
@@ -125,18 +161,26 @@ func (cs *eventStoreChangeSet) stageSave(
 	}
 
 	cs.items = append(cs.items, item)
-
 	return next
 }
 
 // eventStoreDatabase contains data that is committed to the event store.
 type eventStoreDatabase struct {
-	items []*eventstore.Item
+	items      []*eventstore.Item
+	messageIDs map[string]uint64
```

Comment:
```suggestion
	offsets map[string]uint64
```

Makes more sense to name these after the value they store, IMO, just as `items` is.